sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -1199,6 +1199,82 @@ case class Substring(str: Expression, pos: Expression, len: Expression)
 }
 
 /**
+ * Returns the rightmost n characters from the string.
+ */
+@ExpressionDescription(
+  usage = "_FUNC_(str, len) - Returns the rightmost `len` characters from the string `str`.",
+  extended = """
+    Examples:
+      > SELECT _FUNC_('Spark SQL', 3);
+       SQL
+  """)
+case class Right(str: Expression, len: Expression)
+  extends BinaryExpression with ImplicitCastInputTypes with NullIntolerant {
```

Comment:
`Left` has been implemented by substring, but `Right` implemented by substring may be not  very good.
case class Right(str: Expression, len: Expression, child: Expression)
  extends RuntimeReplaceable {
  def this(str: Expression, len: Expression) = {
    this(str, len, Substring(str, If(LessThanOrEqual(len, Literal(0)),
      Literal(Integer.MAX_VALUE), UnaryMinus(len)), len))
  }

  override def flatArguments: Iterator[Any] = Iterator(str, len)

  override def sql: String = s"$prettyName(${str.sql}, ${len.sql})"
}